cmake/run-clang-format.py

```diff
@@ -1,4 +1,18 @@
-#!/usr/bin/env python
+#!/bin/sh
+
+# This file is bilingual. The following shell code finds our preferred python.
+# Following line is a shell no-op, and starts a multi-line Python comment.
+# See https://stackoverflow.com/a/47886254
+""":"
+# prefer python3, then python, then python2
+for cmd in python3 python python2; do
```

Comment:
Cool!  Will this run the script (up to) 3 times? Or does it somehow stop after the first success?

Reply:
Also, it would be good to check if this (still) works on Windows.

(It's not clear if it used to though).

Reply:
If there's a success, the `exec` shell builtin will destroy/overwrite the `sh` process with the Python process.

Windows doesn't use shebangs so it should just treat the [entire block](https://github.com/LLNL/blt/pull/385/files#diff-e622cd37cc768918c8027879b8bcdf07R1-R15) as a comment and use whatever version of Python has a file association configured for `.py` files.